worlds/dark_souls_3/__init__.py

```diff
@@ -1239,49 +1264,19 @@ def write_spoiler(self, spoiler_handle: TextIO) -> None:
                 f"{self.multiworld.get_player_name(self.player)}'s world\n")
 
 
-    def pre_fill(self) -> None:
-        # Fill this manually so that, if very few slots are available in Cemetery of Ash, this
-        # doesn't get locked out by bad rolls on the next two fills.
-        if self.yhorm_location.name == 'Iudex Gundyr':
-            self._fill_local_item("Storm Ruler", {"Cemetery of Ash"},
-                                  lambda location: location.name != "CA: Coiled Sword - boss drop",
-                                  mandatory = True)
-
-        # Don't place this in the multiworld because it's necessary almost immediately, and don't
-        # mark it as a blocker for HWL because having a miniscule Sphere 1 screws with progression
-        # balancing.
-        self._fill_local_item("Coiled Sword", ["Cemetery of Ash", "Firelink Shrine"])
-
-        # If upgrade smoothing is enabled, make sure one raw gem is available early for SL1 players
-        if self.options.smooth_upgrade_items:
-            self._fill_local_item("Raw Gem", [
-                "Cemetery of Ash",
-                "Firelink Shrine",
-                "High Wall of Lothric"
-            ])
-
-
     def _fill_local_item(
         self, name: str,
         regions: List[str],
         additional_condition: Optional[Callable[[DarkSouls3Location], bool]] = None,
-        mandatory = False,
     ) -> None:
         """Chooses a valid location for the item with the given name and places it there.
         
         This always chooses a local location among the given regions. If additional_condition is
         passed, only locations meeting that condition will be considered.
 
-        If mandatory is True, this will throw an error if the item could not be filled in.
+        If the item could not be placed, it will be added to starting inventory.
         """
-        item = next(
-            (
-                item for item in self.multiworld.itempool
-                if item.player == self.player and item.name == name
-            ),
-            None
-        )
-        if not item: return
```

Comment:
I think we should keep this. I don't want to rely on `itempool.remove()` doing the right thing for a _different_ item object that looks the same. It's much safer to pull an existing item out of the pool.